src/sqlfluff/core/rules/base.py

```diff
@@ -432,6 +504,17 @@ class BaseRule:
     _adjust_anchors = False
     targets_templated = False
 
+    # Lint loop / crawl behavior. When appropriate, rules can (and should)
```

Comment:
These are new rule behaviors that rules can leverage to make linting faster. The rules will be updated in a separate PR.